src/popup/components/InputField.vue

```diff
@@ -128,59 +148,62 @@ export default {
       default: null,
     },
   },
-  data: () => ({
-    focused: false,
-  }),
-  computed: {
-    inputId() {
-      return `input-${this._uid}`;
-    },
-    inputMode() {
-      return this.type === 'number' ? 'decimal' : 'text';
-    },
-    hasError() {
-      if (typeof this.message === 'object') {
-        return this.message?.status === 'error';
-      }
-      return !!this.message;
-    },
-    hasWarning() {
-      if (typeof this.message === 'object') {
-        return this.message?.status === 'warning';
-      }
-      return false;
-    },
-    showMessage() {
-      if (typeof this.message === 'object') {
-        return !this.message?.hideMessage;
-      }
-      return !!this.message;
-    },
-    availableTextLimit() {
-      return (this.textLimit && this.value?.length)
-        ? this.textLimit - this.value.length
-        : this.textLimit;
-    },
-  },
-  methods: {
-    checkIfNumber(event) {
+  setup(props, { emit }) {
+    const focused = ref(false);
+
+    const inputId = `input-${getCurrentInstance()?.uid}`;
+    const inputMode = computed(() => props.type === 'number' ? 'decimal' : 'text');
+    const hasError = computed(
+      () => (typeof props.message === 'object')
+        ? props.message?.status === 'error'
+        : !!props.message,
+    );
+    const hasWarning = computed(
+      () => (typeof props.message === 'object')
+        ? props.message?.status === 'warning'
+        : false,
+    );
+    const showMessage = computed(
+      () => (typeof props.message === 'object')
+        ? !props.message?.hideMessage
+        : !!props.message,
```

Comment:
Just a suggestion, but `typeof props.message === 'object'` is used three times in highlighted scope - what do you think of making a computed property that will check that?

Reply:
Yeah, that's true. Unfortunately TypeScript is complaining if I move the test into computed property because TS is not aware of the reactive nature of the computed properties and how this affects if statements :/
But I simplified those conditionals.

Reply:
I tried again and I came up with a solution. Please see how I did it.